### DISCUSSION TITLE
Having trouble connecting Crazyflie with RangeDeck and FlowDeck

### DISCUSSION THREAD

**Discussion by SangjunBae** [bitcraze/discussions#1414] [Q&A]
Hello all,

I'm a complete beginner with Crazyflie and I'm currently testing it out using Flowdeck and RangeDeck (python 3.9 or 3.11 on M1 Macbook Pro).

However, I'm having a major issue with it - it's constantly struggling to connect! Sometimes it works, but other times it just won't, and I have to do something to various troubleshooting methods like restarting python or unplugging and replugging the crazyflie.

Is there a specific sequence or procedure I need to follow to ensure a stable connection? I'm pretty sure I've never had it connect successfully on the first try.

This connection issue is really hindering my progress. If anyone has any helpful links or knows how to fix this, I would be incredibly grateful for your assistance.

Thanks in advance!!!

**Comment by gemenerik** [comment]
Is there anything that could interfere with the connection? Think of physical interference (e.g. concrete floors) or congested radio.

Can you [try a different radio channel](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/#firmware-configuration)?

**Reply by SangjunBae** [reply]
Thanks for your response.

The issue was resolved after replacing the Crazyflie with a new one. I couldn't pinpoint the exact problem. I noticed that the problematic Crazyflie works fine when connected alone, but doesn't work well when a deck is attached. Sometimes it works, and sometimes it doesn't. Do you have any methods or routines to check such issues? It would be helpful to know if you do.

Also, if you have any suggestions to resolve this issue, please let me know.

Thanks in advance!

**Reply by gemenerik** [reply]
It would be good to determine whether it is a hardware or software problem. First make sure you are running all latest release firmware, client and lib, make sure it is unmodified. Also on the decks.
Make sure the decks are properly physically connected. Are all pins coming through on both CF and the deck?
Make sure the attached decks are [compatible](https://www.bitcraze.io/documentation/system/platform/cf2-expansiondecks/#compatibility-matrixes)
Have you tried with another battery?

If you can try to specify the relation between having decks attached and the difficulty with connecting a bit more that would help me out. For example, what decks have you tried to attach? And in what cases has it (not) worked? Is there any pattern there?
